### ISSUE TITLE
Inconistent result after applying Firewall Rule

### ISSUE THREAD

**Issue by ejstreet** [civo/terraform-provider-civo#36]
I get the following error when trying to add firewall rules to a firewall in terraform:
```
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to civo_firewall_rule.network["inbound_http"],
│ provider "provider[\"registry.terraform.io/civo/civo\"]" produced an
│ unexpected new value: Root resource was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```
Usually the first rule in a map will show up in the webUI, even though it does produce this error.

My config:
```
terraform { 
    required_providers {
        civo = {
            source = "civo/civo"
            version = "~> 0.10.2"
        }
    }
}

provider "civo" {
    region = var.region
}

resource "civo_network" "network" {
    label  = var.network_name
    region = var.region
}

resource "civo_firewall" "network" {
    name       = var.network_name
    network_id = civo_network.network.id
}

resource "civo_firewall_rule" "network" {
    firewall_id = civo_firewall.network.id
    region      = var.region
    
    for_each = var.firewall_rules

    label      = each.key
    protocol   = each.value.protocol
    start_port = each.value.start_port
    end_port   = each.value.end_port
    cidr       = each.value.cidr
    direction  = each.value.direction
}

variable "region" {
    description = "The region in which the network will be deployed."
    type        = string
}

variable "network_name" {
    description = "Desired name of the network."
    type        = string
}

variable "firewall_rules" {
    description = "Map of firewall rules. The key is used as the label. See https://registry.terraform.io/providers/civo/civo/latest/docs/resources/firewall_rule for details."
    type        = map(object({
        protocol   = string
        start_port = number
        end_port   = number
        cidr       = list(string)
        direction  = string
    }))
    default = {}
}
```
Inputs 
```
region       = "LON1"
network_name = "main"

firewall_rules = {
  inbound_http = {
    protocol = "tcp"
    start_port = 80
    end_port = 80
    cidr = ["0.0.0.0/0"]
    direction = "ingress"
  },
}

**Comment by ejstreet** [comment]
I imported the created rule and ran `plan`, it gave me this:
```
Terraform will perform the following actions:

  # civo_firewall_rule.network["inbound_http"] must be replaced
-/+ resource "civo_firewall_rule" "network" {
      + end_port    = "80" # forces replacement
      ~ id          = "febb14f0-a80c-4b92-9e8f-165b269ab4a7" -> (known after apply)
        # (6 unchanged attributes hidden)
    }
```

**Comment by ejstreet** [comment]
I had a go at building the provider locally with 

func resourceFirewallRule() *schema.Resource {```
...
end_port": {				
				Optional:     true
```
However this did not fix the issue. Unfortunately I have next to no experience with Go, so I don't really know what to do next to fix the issue.

**Comment by alejandrojnm** [comment]
OK i will check this

**Comment by alejandrojnm** [comment]
This is fixed in the v0.10.4
